src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -22,6 +25,28 @@ internal sealed class TaskCenterSolutionAnalysisProgressReporter
         private TaskCompletionSource<VoidResult> _currentTask;
         private DateTimeOffset _lastTimeReported;
 
+        private int _lastPendingItemCount;
+        private ProgressStatus _lastProgressStatus;
+        private ProgressStatus _lastShownProgressStatus;
```

Comment:
basically, the appraoch i recommend woudl allow you do get rid of this entirely :)   And you wouldn't need logic around do you need to update the UI.  You just always update the UI at a reasonable clip (with appropraite batching/pauses) after the last notification you get, and the UI always represents the current state of the system as you read it at that time.

You always reach a fixed point becuse the UI always updates reading the latest data, and any time data change the UI will always know it needs to update.  

it's a simpler and eaier system (and also what we use in a bunch of other places to ensure no races, and to also decouple this sort of thing).

Thanks!

Reply:
due to the interval. that most of the time just show "Paused". pull model only work for the count. for evaluating and pausing flipping in high frequency. the "paused" will be the dominant one. 

you can try and if you can make it work with simple pull mode. I am happy to review and accept the PR!